Update tests to use node:test, remove tap

bencoder opened the pull request:
Updates tests to use node:test
Remove tap
Converted tests to async/await style over callbacks and removed `t.plan(x)` statements where appropriate

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/fastify/fastify/blob/master/CONTRIBUTING.md

By making a contribution to this project, I certify that:

* (a) The contribution was created in whole or in part by me and I
  have the right to submit it under the open source license
  indicated in the file; or

* (b) The contribution is based upon previous work that, to the best
  of my knowledge, is covered under an appropriate open source
  license and I have the right under that license to submit that
  work with modifications, whether created in whole or in part
  by me, under the same open source license (unless I am
  permitted to submit under a different license), as indicated
  in the file; or

* (c) The contribution was provided directly to me by some other
  person who certified (a), (b) or (c) and I have not modified
  it.

* (d) I understand and agree that this project and the contribution
  are public and that a record of the contribution (including all
  personal information I submit with it, including my sign-off) is
  maintained indefinitely and may be redistributed consistent with
  this project or the open source license(s) involved.
-->

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

bencoder committed:
Update tests to use node:test

Uzlopak commented:
Yeah the windows fails. Maybe simplest solution: change file extensions of all test files from `.js` to `.test.js`

gurgunday reviewed (changes requested):
Hi, thanks a lot for the PR!

Please:

1) Install `c8`
2) Change the `test:unit` command to `c8 --100 node --test`
3) As @Uzlopak said, for consistency, it would be great if you could add `.test.` to the test files

bencoder committed:
Rename all tests and update to use c8 with 100% coverage check, fix invalid test

dancastillo reviewed (commented):
Can you add back the `t.plan` to the tests `node:test` does support test count assertions [ref](https://nodejs.org/api/test.html#contextplancount)

bencoder commented:
@dancastillo I've removed them where they are not adding anything - they only really make sense when you have assertions in callbacks or conditional statements - everywhere I've removed them has been where I've changed to using `await` so the assertions will always run and setting `t.plan` just seems like extra friction. 

If they should be a standard for fastify tests though then I can add them back if necessary

dancastillo commented:
@bencoder That makes sense to me! As for the standard with Fastify tests, I believe we have them in place for most of our tests but like you mention likely for callbacks or conditionals

I'll defer it to @Uzlopak or @gurgunday to confirm whether it's a strict requirement or is okay

jsumners commented:
I would reserve plans for tests of asynchronous operations.

gurgunday reviewed (approved):
lgtm

dancastillo commented on test/access.test.js:
```diff
@@ -93,35 +90,32 @@ t.test('Get all data that we set in session', t => {
-      })
-    })
+  })
+  t.assert.ifError(postResponse.error)
```
```suggestion
  t.assert.ok(postResponse)
```

dancastillo commented on test/access.test.js:
```diff
@@ -93,35 +90,32 @@ t.test('Get all data that we set in session', t => {
+      cookie: postResponse.headers['set-cookie']
+    }
+  })
+  t.assert.ifError(getResponse.error)
```
```suggestion
  t.assert.ok(getResponse)
```

dancastillo reviewed (commented):
Looks like there is no `error` defined on `await fastify.inject(...)` response [ref](https://github.com/fastify/light-my-request/blob/master/types/index.d.ts#L78-L93)

So this below will never fail. 
```js
t.assert.ifError(response.error)
```

Can we update all similar asserts to below to validate we received a truthy response
```js
  t.assert.ok(response)
```

bencoder committed:
Change all instances where we are checking `ifError(response.error)` to `ok(response)` because response.error does not exist

dancastillo reviewed (approved):
LGTM
